packages/migrations/src/migrateDockerNetworkIpRange/connectContainersToNetworkWithPrio/connectContainerWithIp.ts

```diff
@@ -55,34 +56,73 @@ export async function connectContainerWithIp({
 
   // check target container is running, otherwise the docker network connect might not take effect
   const targetContainer = docker.getContainer(containerName);
-  const containerInfo = await targetContainer.inspect();
-  if (
-    !containerInfo.State.Running &&
-    containerName !== params.dappmanagerContainerName
-  ) {
-    logs.warn(`container ${containerName} is not running, restarting it`);
-    await targetContainer.restart();
-  }
+  try {
+    // There has been edge cases where docker containers are in an intermedium state with a different docker container name
+    // i.e b7903a289091_DAppNodeCore-bind.dnp.dappnode.eth instead of DAppNodeCore-bind.dnp.dappnode.eth
+    const containerInfo = await targetContainer.inspect();
+    if (
+      !containerInfo.State.Running &&
+      containerName !== params.dappmanagerContainerName
+    ) {
+      logs.warn(`container ${containerName} is not running, restarting it`);
+      await targetContainer.restart();
+    }
+
+    const hasContainerRightIp =
+      removeCidrSuffix(aliasesIpsMap.get(containerName)?.ip || "") ===
+      containerIp;
 
-  const hasContainerRightIp =
-    removeCidrSuffix(aliasesIpsMap.get(containerName)?.ip || "") ===
-    containerIp;
+    if (hasContainerRightIp)
+      logs.info(
+        `container ${containerName} has right IP and is connected to docker network`
+      );
+    else {
+      logs.info(
+        `container ${containerName} does not have right IP and/or is not connected to docker network`
+      );
+      await connectContainerRetryOnIpUsed({
+        network,
+        containerName,
+        maxAttempts: 20,
+        ip: containerIp,
+        aliasesIpsMap,
+      });
+    }
+  } catch (e) {
+    // check if container does not exist 404
+    if (containerName === params.bindContainerName && e.statusCode === 404) {
+      logs.warn(`container ${containerName} does not exist`);
+      // check if there are intermedium containes with different names and remove them, then do docker compose up of the container
+      // look for the intermedium container by looking for any docker container that has in its name the container name.
+      try {
+        const containers = await docker.listContainers({
+          all: true,
+        });
+        // check if there are any intermedium containers
+        const intermediumContainer = containers.find((container) =>
+          container.Names.some((name) => name.includes(containerName))
+        );
+        if (intermediumContainer) {
```

Comment:
Just an idea: Maybe we could wait for a bit until docker removes this renamed container before frocefully removing it ourselves?